pkg/pipeline/pipeline_builder.go

```diff
@@ -216,7 +215,7 @@ func getIngester(params config.StageParam) (ingest.Ingester, error) {
 	case "file", "file_loop", "file_chunks":
 		ingester, err = ingest.NewIngestFile(params)
 	case "collector":
-		ingester, err = ingest.NewIngestCollector(params)
+		ingester, err = ingest.NewIngestCollector(params.Ingest.Collector)
```

Comment:
If we make this `New` ingest procedure specific, we should consider making all the other `New` procedures specific. If not,then this one should also be generic (`params` instead of `params.Ingest.Collector` for consistency.